src/components/recordedit/form-container.tsx

```diff
@@ -216,74 +335,204 @@ const FormRow = ({ columnModelIndex }: FormRowProps): JSX.Element => {
-      }
-    </>)
-  }
+        {typeof formIndex === 'number' && formIndex in showPermissionError && (
+          <div
+            className={`column-permission-warning column-permission-warning-${safeClassNameId}`}
+          >
+            {permissionError}
+          </div>
+        )}
+      </>
+    );
+  };
 
   return (
-    <div className={`form-inputs-row ${showSelectAll ? 'highlighted-row' : ''}`} ref={container}>
+    <div
+      className={`form-inputs-row ${showSelectAll ? 'highlighted-row' : ''}`}
+      ref={container}
+    >
       <div className='inputs-row'>
         {forms.map((formNumber: number, formIndex: number) => (
-          <div key={`form-${formNumber}-input-${columnModelIndex}`} className='entity-value'>
+          <div
+            key={`form-${formNumber}-input-${columnModelIndex}`}
+            /**
+             * This is added to show the form is selected to apply the change when it is in edit mode
+             */
+            className={`form-overlay entity-value ${activeForm.includes(formNumber) && showSelectAll
+              ? 'entity-active'
+              : ''
+              }`}
+            onClick={() => {
+              // I couldn’t test that scenario, since on load we’re removing the forms that user cannot edit,
+              if (
+                appMode === appModes.EDIT &&
+                canUpdateValues &&
+                !canUpdateValues[`${formNumber}-${cm.column.name}`]
+              ) {
+                return;
+              }
+              handleFormClick(formNumber);
+            }}
+          >
             {renderInput(formNumber, formIndex)}
           </div>
         ))}
       </div>
-      {showSelectAll && <SelectAllRow columnModelIndex={columnModelIndex} />}
+      {showSelectAll && (
+        <SelectAllRow
+          activeForm={activeForm}
+          setActiveForm={setActiveForm}
+          columnModelIndex={columnModelIndex}
+        />
+      )}
     </div>
-  )
-
+  );
 };
 
 /**
  * shows the select all row
  * NOTE this is its own component to avoid rerendering the whole row on each change.
  */
-const SelectAllRow = ({ columnModelIndex }: FormRowProps) => {
+const SelectAllRow = ({
+  columnModelIndex,
+  activeForm,
+  setActiveForm,
+}: FormRowProps) => {
   const {
-    columnModels, forms, reference, waitingForForeignKeyData, foreignKeyData, appMode,
-    canUpdateValues, toggleActiveSelectAll, logRecordeditClientAction
+    columnModels,
+    forms,
+    reference,
+    waitingForForeignKeyData,
+    foreignKeyData,
+    appMode,
+    canUpdateValues,
+    toggleActiveSelectAll,
+    logRecordeditClientAction,
   } = useRecordedit();
+  const {
+    watch,
+    reset,
+    getValues,
+    formState: { errors },
+  } = useFormContext();
+  let ref = useRef(null);
+  const [isEmpty, setIsEmpty] = useState(true);
+  const [selected, setSelected] = useState(false);
```

Comment:
What does this state variable mean? Please comment and also use a better name. Is this supposed to capture if all are selected or not?